modules/modules.go

```diff
@@ -20,6 +20,9 @@ var (
 	// modulesLocked locks `modules` during starting.
 	modulesLocked = abool.New()
 
+	sleepMode                         = abool.NewBool(false)
+	taskSchedulerSleepModeExitChannel = make(chan struct{})
```

Comment:
I would recommend to call this `wakeTaskScheduler` and create it with a buffer of one. When writing to it, always use a select with a default, so multiple can wake the task scheduler. Put this in a function to make it harder to get it wrong in the future.